source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -755,10 +755,10 @@ know which MongoDB versions will be compatible or incompatible in the future.
 Verifying setName with TopologyType Single
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-A client MAY allow the user to supply a setName
-with an initial TopologyType of Single.
-In this case, if the ServerDescription's setName is null or wrong,
-the client MUST throw an error on every operation.
+A client MAY allow the user to supply a setName with an initial TopologyType
```

Comment:
Should we explicitly say that the server's connection pool must be cleared when this happens?

Reply:
Why would it need to be cleared?  It's a configuration error, but the sockets are not necessarily in a bad state as a result.

Reply:
Looking through the sdam spec for when connection pools are cleared, my impression is we do this when we expect the server to have closed the connections for some reason, to avoid having all of those connections error when we try to use them.

The only way a server can change its set name is if it is restarted, therefore, applying similar logic it seems reasonable to clear connection pool if set name is wrong.

If the first connection to a server yields wrong set name, there shouldn't be any connections established to it and whether pool is cleared shouldn't matter.